WordPress/Classes/ViewRelated/Plans/PlanDetailViewModel.swift

```diff
@@ -41,20 +41,22 @@ struct PlanDetailViewModel {
         }
     }
 
-    var noResultsViewModel: WPNoResultsView.Model? {
+    var noResultsViewModel: NoResultsViewController.Model? {
         switch features {
         case .loading:
-            return WPNoResultsView.Model(
-                title: NSLocalizedString("Loading Plan...", comment: "Text displayed while loading plans details")
-            )
+            return NoResultsViewController.Model(title: NSLocalizedString("Loading Plan...", comment: "Text displayed while loading plans details"))
         case .ready:
             return nil
         case .error:
-            return WPNoResultsView.Model(
-                title: NSLocalizedString("Oops", comment: ""),
-                message: NSLocalizedString("There was an error loading the plan", comment: ""),
-                buttonTitle: NSLocalizedString("Contact support", comment: "")
-            )
+            let appDelegate = WordPressAppDelegate.sharedInstance()
+            if (appDelegate?.connectionAvailable)! {
+                return NoResultsViewController.Model(title: NSLocalizedString("Oops", comment: ""),
+                                                     subtitle: NSLocalizedString("There was an error loading the plan", comment: "Text displayed when there is a failure loading the plan details"),
+                                                     buttonText: NSLocalizedString("Contact support", comment: "Button label for contacting support"))
+            } else {
+                return NoResultsViewController.Model(title: NSLocalizedString("No connection", comment: ""),
+                                                     subtitle: NSLocalizedString("An active internet connection is required to view plans", comment: ""))
+            }
```

Comment:
Just for cleanness: could we here create a helper method that returns the `errorModel`? 
Something like `return errorViewModel()`

Reply:
The helper method was going to be almost as bad, so I moved the strings to a struct to make it more readable.